frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -322,15 +339,41 @@ function isParametersWidgetContainersSticky(parameterCount: number) {
 }
 
 function getParametersListComponent({
-  isEmbeddingSdk,
+  isQuestion,
   isDashboard,
+  isEmbeddingSdk,
 }: {
+  isQuestion: boolean;
+  isDashboard: boolean;
   isEmbeddingSdk: boolean;
+}) {
+  return shouldSyncUrlParameters({ isQuestion, isDashboard, isEmbeddingSdk })
+    ? SyncedParametersList
+    : ParametersList;
+}
+
+function shouldSyncUrlParameters({
+  isQuestion,
+  isDashboard,
+  isEmbeddingSdk,
+}: {
+  isQuestion: boolean;
   isDashboard: boolean;
+  isEmbeddingSdk: boolean;
 }) {
+  // Couldn't determine if it's a question or a dashboard until one becomes true.
+  if (!isQuestion && !isDashboard) {
```

Comment:
Without this guard condition, when both are false, it would fall in the else block which means it's gonna sync URL when we still couldn't determine if we want the URL to be synced yet.